src/engine/runtime.js

```diff
@@ -1233,12 +1233,14 @@ class Runtime extends EventEmitter {
     /**
      * Update an edge-activated hat block value.
      * @param {!string} blockId ID of hat to store value for.
+     * @param {!string} threadTargetId Target ID for the thread that the block belongs to
      * @param {*} newValue Value to store for edge-activated hat.
      * @return {*} The old value for the edge-activated hat.
      */
-    updateEdgeActivatedValue (blockId, newValue) {
-        const oldValue = this._edgeActivatedHatValues[blockId];
-        this._edgeActivatedHatValues[blockId] = newValue;
+    updateEdgeActivatedValue (blockId, threadTargetId, newValue) {
+        const blockAndTargetId = `${blockId}_${threadTargetId}`;
```

Comment:
Since this information is now target specific should it be moved onto the Target? I mention this because that would be faster than creating this string for hat blocks that run every frame tick and because it seems weird to have Target specific data on the runtime.

Reply:
Yeah, @paulkaplan and I were also talking about that earlier. I think that makes sense and is an easy enough/low risk change.